pkg/stanza/fileconsumer/scanner.go

```diff
@@ -35,11 +37,22 @@ func NewPositionalScanner(r io.Reader, maxLogSize int, startOffset int64, splitF
 		Scanner: bufio.NewScanner(r),
 	}
 
-	buf := make([]byte, 0, 16384)
+	bufSize := defaultBufSize
+	if maxLogSize < bufSize {
+		bufSize = maxLogSize
+	}
+
+	buf := make([]byte, 0, bufSize)
```

Comment:
I'm not sure this change is necessary. The documentation for [`Scanner.Buffer(buf, max)`](https://pkg.go.dev/bufio#Scanner.Buffer) says: `The maximum token size is the larger of max and cap(buf). If max <= cap(buf), Scan will use this buffer only and do no allocation.`

Reply:
If `max_log_size` < 16384, the Scan may emit log entries larger than `max_log_size` without this change.